server/fleet/scripts.go

```diff
@@ -194,7 +230,7 @@ type HostScriptResult struct {
 
 	// TeamID is only used for authorization, it must be set to the team id of
 	// the host when checking authorization and is otherwise not set.
-	TeamID *uint `json:"team_id" db:"-"`
+	TeamID *uint `json:"team_id" db:"-"` // TODO: should we omit this from the json result?
```

Comment:
Note that it has to be marshalled as `json` so that it gets passed to `rego` for authorization. It doesn't get returned (IIRC) because we only set the team ID in the struct created for authorization, not for the actual API response.

Reply:
Oh I see now that it doesn't have the `omitempty` which may be what you were referring to, it does make it to the response, but null... Hadn't noticed that, we should probably fix this.

Reply:
Though this can certainly wait for after the freeze.

Reply:
> Note that it has to be marshalled as json so that it gets passed to rego for authorization. It doesn't get returned (IIRC) because we only set the team ID in the struct created for authorization, not for the actual API response.

Ah, thanks for the additional context! I was getting thrown off by it in my dev testing of the API. I'll go ahead and merge and we can look into further during the freeze.

Reply:
Yeah I agree that it is surprising and could be error-prone. Might be worth using a distinct struct for authorization.